### PR TITLE
refactor/ `GatewayEVMAMM` to use `ClientOrderTracker` and new `InFlightOrder`

### PR DESCRIPTION
**Pull request by CrimsonJacket**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:

[sc-24379] [sc-24423] [sc-26652]

**Tests performed by the developer**:

**Tips for QA testing**:

### CODE REVIEW

File: hummingbot/connector/gateway_in_flight_order.py [hunk 1]
```diff
@@ -19,58 +16,24 @@ def __init__(self,
                  price: Decimal,
                  amount: Decimal,
                  creation_timestamp: float,
-                 gas_price: Decimal,
-                 initial_state: str = "PENDING_CREATE"):
+                 gas_price: Optional[Decimal] = Decimal("0"),
+                 initial_state: str = OrderState.PENDING_CREATE,
+                 is_approval: bool = False):
```
Comment (vic-en) [inline hummingbot/connector/gateway_in_flight_order.py:21]:
```suggestion
```
I see `is_approval` isn't really used in this class. Let us clean it up here and in initializing calls.

**Review by vic-en — CHANGES_REQUESTED** [review]

File: hummingbot/connector/gateway_in_flight_order.py (hunk 1 above)
Comment (CrimsonJacket) [reply hummingbot/connector/gateway_in_flight_order.py:21]: It is used in line 31

**Comment by PtrckM** [comment]
- getting error during filled trade event side of  uniswap or avalanche

image
image

```
2022-04-19 06:20:21,002 - 18559 - GatewayEVMAMM - ERROR - 'bool' object is not callable
Traceback (most recent call last):
  File "/home/ubuntu/ordertracker/hummingbot/connector/gateway_EVM_AMM.py", line 840, in _status_polling_loop
    self.update_canceling_transactions(self.canceling_orders),
  File "/home/ubuntu/ordertracker/hummingbot/connector/gateway_EVM_AMM.py", line 185, in canceling_orders
    for cancel_order in self.amm_orders
  File "/home/ubuntu/ordertracker/hummingbot/connector/gateway_EVM_AMM.py", line 175, in amm_orders
    return [
  File "/home/ubuntu/ordertracker/hummingbot/connector/gateway_EVM_AMM.py", line 178, in <listcomp>
    if in_flight_order.is_open() and not in_flight_order.is_approval_request
TypeError: 'bool' object is not callable
2022-04-19 06:20:21,002 - 18559 - hummingbot.core.utils.async_utils - ERROR - Unhandled error in background task: 'bool' object is not callable
Traceback (most recent call last):
  File "/home/ubuntu/ordertracker/hummingbot/core/utils/async_utils.py", line 9, in safe_wrapper
    return await c
  File "/home/ubuntu/ordertracker/hummingbot/strategy/amm_arb/amm_arb.py", line 219, in apply_gateway_transaction_cancel_interval
    await gateway.cancel_outdated_orders(self._gateway_transaction_cancel_interval)
  File "/home/ubuntu/ordertracker/hummingbot/connector/gateway_EVM_AMM.py", line 948, in cancel_outdated_orders
    o for o in self.amm_orders
  File "/home/ubuntu/ordertracker/hummingbot/connector/gateway_EVM_AMM.py", line 175, in amm_orders
    return [
  File "/home/ubuntu/ordertracker/hummingbot/connector/gateway_EVM_AMM.py", line 178, in <listcomp>
    if in_flight_order.is_open() and not in_flight_order.is_approval_request

```

- config, logs, csv and sqlite: 
[stuff.zip](https://github.com/CoinAlpha/hummingbot/files/8510555/stuff.zip)

File: hummingbot/connector/gateway_in_flight_order.py (hunk 1 above)
Comment (vic-en) [reply hummingbot/connector/gateway_in_flight_order.py:21]: I still see it's redundant. Your changes in `hummingbot/connector/gateway_EVM_AMM.py` already send the appropraite `initial_state`.

**Review by vic-en — CHANGES_REQUESTED** [review]

File: hummingbot/connector/gateway_in_flight_order.py (hunk 1 above)
Comment (CrimsonJacket) [reply hummingbot/connector/gateway_in_flight_order.py:21]: Ahh I see what you mean. Ok

**Review by vic-en — APPROVED** [review]
Changes LGTM

**Comment by CrimsonJacket** [comment]
This should resolve Bug [sc-26652]

**Comment by CrimsonJacket** [comment]
Recreated PR in PR #236

**Pull request closed without merging**